src/code-generation/LuaCodeGenerator.ts

```diff
@@ -0,0 +1,159 @@
+import _ from 'lodash'
+import CodeGeneratorBase, { CodeGeneratorBaseSettings } from './CodeGeneratorBase'
+import { CodeGenNative, CodeGenParam, CodeGenType } from './ICodeGenerator'
+
+export interface LuaCodeGeneratorSettings extends CodeGeneratorBaseSettings {
+  generateComments  : boolean
+  useNativeTypes    : boolean
+  cppCompliant      : boolean
+  includes          : string[]
+  invokeFunction    : string
+  invokeSupportsVoid: boolean
+  oneLineFunctions  : boolean
+  includeNdbLinks   : boolean
```

Comment:
Are these settings applicable for Lua? (Especially the `cppCompliant` one)

Reply:
cppCompliant is not, the rest are. Invoke Supports Void, some don't have that registered etc.
One line functions also apply.
Normally, LUA does not have types, however, we still need to indicate the required typing.